### ISSUE TITLE
Speculative plans from Pull Requests

### ISSUE THREAD

**Issue by akozhuharov** [terrakube-io/terrakube#1011]
### Feature description 💡

TLDR:
What does the community think about adding the option to run speculative plans based on if the VCS event is from a pull/merge/etc request?

Details:
Currently Terrakube only runs on specific VCS branches, unless my understanding is incomplete. It would be very useful to see results of changes prior to merging them to the main trunk of corresponding repos. 

For implementation I would go with another field in the Workspaces settings with a corresponding boolean flag in the Database. Then per VCS plugin the [event](https://github.com/AzBuilder/terrakube/blob/341b6e6bab8fd2a6a35001e795aec97d1f378d55/api/src/main/java/org/terrakube/api/plugin/vcs/provider/github/GitHubWebhookService.java#L44) can be handled

**Comment by alfespa17** [comment]
You are correct for now we only run the jobs during Push events, it would be nice to support other events like pull request for example to run a plan for example

**Comment by juan-vg** [comment]
I believe this is closely related to #396

**Comment by alfespa17** [comment]
At least with github you can have an "alternative", if you setup the workspace webhook like the following:

![image](https://github.com/user-attachments/assets/02ac034a-01ac-4501-8b74-2c960661b41f)

Terrakube will update the commit status in github when you trigger a push in branch called "feature" for example

> the branch name can be a regex or multiple regex with ","

![image](https://github.com/user-attachments/assets/875f443b-069b-4a2a-9dab-47a9aca2adb2)

And when you create PR to "main" you will see something like this inside the checks

![image](https://github.com/user-attachments/assets/b7abaad8-8cb9-4f19-80ad-42ec3d45103b)

**Comment by hirenko-v** [comment]
@alfespa17 in this case Terrakube will run a plan for each push to the branches. We probably don't want to run the plan before PR is opened

**Comment by alfespa17** [comment]
> @alfespa17 in this case Terrakube will run a plan for each push to the branches. We probably don't want to run the plan before PR is opened

That is correct @hirenko-v for now it only support "push events" it still requier some internal changes to support the "pull request event"

**Comment by juan-vg** [comment]
Since it can be done via repository CI, I personally don't believe it's a priority but a nice-to-have. The only requirement left is to be able to provide the branch when triggering the jobs via API, and this was already enabled by #1635 which will be included in the next release I believe. Just my two cents

**Comment by alfespa17** [comment]
I am closing this issue, terrakube now support push, pull request and release events for gitlab and github, I don't see people using Terrakube with bitbucket or azure devops so I think this can be close now
